Library/Homebrew/livecheck/strategy/apache.rb

```diff
@@ -73,7 +73,7 @@ def self.find_versions(url, regex, cask: nil, &block)
           # * `/href=["']?example-v?(\d+(?:\.\d+)+)-bin\.zip/i`
           regex ||= /href=["']?#{Regexp.escape(match[:prefix])}v?(\d+(?:\.\d+)+)#{Regexp.escape(suffix)}/i
 
-          PageMatch.find_versions(page_url, regex, cask: cask, &block)
+          PageMatch.find_versions(url: page_url, regex: regex, **unused, &block)
```

Comment:
It's not currently necessary and I weighed whether to omit it from the `PageMatch#find_versions` method calls. I included it because I didn't see any notable downside but there are potential benefits:

* Having `**unused` in the `PageMatch#find_versions` call automatically ensures that an unused parameter in the calling strategy would be passed through to `PageMatch#find_versions`. It could be that `PageMatch#find_versions` is modified in the future to include a named `x` parameter but, for whatever reason, it doesn't make sense to add this parameter to the calling strategies. This setup supports this situation without requiring us to update the twelve related strategies. Decoupling `#find_versions` parameters between strategies to ease future development is the general goal of this PR and I would argue that this `**unused` approach is part of it.
* If `**unused` is omitted from the `PageMatch#find_versions` method call, then it's not used in the calling strategy's `#find_versions` method and we have to rename it to `_unused` to pass `brew style`. As mentioned above, if we need to pass through the unused parameters in the future, we'll have to rename `_unused`, update the method signature, and modify the `PageMatch#find_versions` call for twelve strategies. Speaking from experience, it's time-consuming and not fun.

My thinking is that it was easy to take care of this now and it could save us from doing more work in the future (potentially making related development less onerous). At this point, it would take work to modify these strategies to not use `**unused` in the `PageMatch#find_versions` call, so it doesn't seem worth it to me unless there's a compelling reason.

Reply:
> * It could be that `PageMatch#find_versions` is modified in the future to include a named `x` parameter

Do you foresee doing anything like that in the near future? I'm OK with future-proofing code for an upcoming feature, but I'm not a fan of doing so for hypothetical stuff that might never happen. The only upside I see is that it standardizes method signatures.

Anyways, that's not a major issue for me, so let's merge this as is 👍🏻

Reply:
> Do you foresee doing anything like that in the near future? I'm OK with future-proofing code for an upcoming feature, but I'm not a fan of doing so for hypothetical stuff that might never happen. The only upside I see is that it standardizes method calls.

What I have lined up for the near future brings the parameters for `PageMatch#find_versions` and the `#find_versions` methods in strategies that call it into parity and additional changes will maintain it. At the moment, this is more about ensuring that `PageMatch#find_versions` gets all the arguments from the original `#find_versions` call in livecheck (outside of the original `url`, if a strategy generates one internally).

I think it's reasonable to standardize on the idea that when one `#find_versions` method calls another, it should pass along the arguments it was given (outside of values that are overridden by generated ones) instead of assuming that the upstream strategy won't need an `unused` argument. I can understand the YAGNI view when it comes to bigger changes but omitting `**unused` would feel like a bit of an oversight to me.

This won't hurt but it may help, so I'm going to go ahead with merging. If `**unused` becomes an issue in some way, we can always revisit it later.